### PR TITLE
Fix Cache issue

### PR DESCRIPTION
**Pull request by krzysztofzablocki**
Fixes issue #774

### CODE REVIEW

File: Sourcery/Sourcery.swift [hunk 1]
```diff
@@ -342,10 +342,13 @@ extension Sourcery {
     private func load(artifacts: String, contentSha: String) -> FileParserResult? {
         var unarchivedResult: FileParserResult?
         SwiftTryCatch.try({
-                              if let unarchived = NSKeyedUnarchiver.unarchiveObject(withFile: artifacts) as? FileParserResult, unarchived.sourceryVersion == Sourcery.version, unarchived.contentSha == contentSha {
+
+            if let unarchived = NSKeyedUnarchiver.unarchiveObject(withFile: artifacts) as? FileParserResult {
+                if unarchived.sourceryVersion == Sourcery.version, unarchived.contentSha == contentSha {
                                   unarchivedResult = unarchived
```
Comment (Liquidsoul) [inline Sourcery/Sourcery.swift]: Maybe change the indentation for this two lines too?

**Comment by ilyapuchka** [comment]
What was the underlying issue though? I see the name of `Version` changed to `SourceryVersion` but it changed everywhere...

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,10 @@
 # Sourcery CHANGELOG
 
 ---
-## Master
+## 0.16.2
```
Comment (Liquidsoul) [inline CHANGELOG.md]: Usually we update this during the release process 🤔
I would keep the `Master` here and release the `0.16.2` after merging this.

**Comment by krzysztofzablocki** [comment]
@ilyapuchka swiftlint has object called `Version` and depending on how we import our framework vs theirs a different one was used, before this pr all caches used SwiftLint version value instead of Sourcery's

File: SourceryUtils/Sources/Version.swift [hunk 3]
```diff
@@ -8,9 +8,9 @@
 
 import Foundation
 
-public struct Version {
+public struct SourceryVersion {
     public let value: String
-    public static let current = Version(value: inUnitTests ? "Major.Minor.Patch" : "0.16.1")
+    public static let current = SourceryVersion(value: inUnitTests ? "Major.Minor.Patch" : "0.16.2")
```
Comment (Liquidsoul) [inline SourceryUtils/Sources/Version.swift]:
```suggestion
    public static let current = SourceryVersion(value: inUnitTests ? "Major.Minor.Patch" : "0.16.1")
```

**Comment by krzysztofzablocki** [comment]
@Liquidsoul that would be great, thanks!

**Comment by ilyapuchka** [comment]
@krzysztofzablocki waaaat =)))) that's unexpected=)

**Pull request merged**